ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/computation/DeprecatedStatementsCheckerTest.java

```diff
@@ -0,0 +1,269 @@
+    when(metaStore.getSource(TABLE_1)).thenReturn(tableSource1);
+  }
+
+  @Test
+  public void shouldDeprecateStreamStreamJoinsWithNoGrace() {
+    for (final JoinType joinType : JoinType.values()) {
+      checkDeprecatedStatement(String.format(
+          "SELECT * FROM %s AS l "
+              + "%s JOIN %s AS r WITHIN 1 SECOND ON l.K = r.K;",
+          STREAM_1.text(),
+          (joinType == JoinType.OUTER) ? "FULL OUTER" : joinType,
+          STREAM_2.text()
+      ));
+    }
+  }
+
+  @Test
+  public void shouldDeprecateStreamStreamJoinsWithNoGraceOnMultiStreamStreamJoinOrder1() {
+    for (final JoinType joinType : JoinType.values()) {
+      checkDeprecatedStatement(String.format(
+          "SELECT * FROM %s AS l "
+              + "%s JOIN %s AS m WITHIN 1 SECOND GRACE PERIOD 1 SECOND ON l.K = m.K "
+              + "%s JOIN %s AS r WITHIN 1 SECOND ON m.K = r.K;",
+          STREAM_1.text(),
+          (joinType == JoinType.OUTER) ? "FULL OUTER" : joinType,
+          STREAM_2.text(),
+          (joinType == JoinType.OUTER) ? "FULL OUTER" : joinType,
+          STREAM_3.text()
+      ));
+    }
+  }
+
+  @Test
+  public void shouldDeprecateStreamStreamJoinsWithNoGraceOnMultiStreamStreamJoinOrder2() {
+    for (final JoinType joinType : JoinType.values()) {
+      checkDeprecatedStatement(String.format(
+          "SELECT * FROM %s AS l "
+              + "%s JOIN %s AS r WITHIN 1 SECOND ON m.K = r.K "
+              + "%s JOIN %s AS m WITHIN 1 SECOND GRACE PERIOD 1 SECOND ON l.K = m.K;",
+          STREAM_1.text(),
+          (joinType == JoinType.OUTER) ? "FULL OUTER" : joinType,
+          STREAM_2.text(),
+          (joinType == JoinType.OUTER) ? "FULL OUTER" : joinType,
+          STREAM_3.text()
+      ));
+    }
+  }
+
+  @Test
+  public void shouldDeprecateStreamStreamJoinsWithNoGraceOnMultiStreamStreamTableJoinOrder1() {
+    for (final JoinType joinType : JoinType.values()) {
+      checkDeprecatedStatement(String.format(
+          "SELECT * FROM %s AS l "
+              + "%s JOIN %s AS m WITHIN 1 SECOND ON l.K = m.K "
+              + "%s JOIN %s AS r ON l.K = r.K;",
+          STREAM_1.text(),
+          (joinType == JoinType.OUTER) ? "FULL OUTER" : joinType,
+          STREAM_2.text(),
+          (joinType == JoinType.OUTER) ? "FULL OUTER" : joinType,
+          TABLE_1.text()
+      ));
+    }
+  }
+
+  @Test
+  public void shouldDeprecateStreamStreamJoinsWithNoGraceOnMultiStreamStreamTableJoinOrder2() {
+    for (final JoinType joinType : JoinType.values()) {
+      checkDeprecatedStatement(String.format(
+          "SELECT * FROM %s AS l "
+              + "%s JOIN %s AS r ON l.K = r.K "
+              + "%s JOIN %s AS m WITHIN 1 SECOND ON l.K = m.K;",
+          STREAM_1.text(),
+          (joinType == JoinType.OUTER) ? "FULL OUTER" : joinType,
+          STREAM_2.text(),
+          (joinType == JoinType.OUTER) ? "FULL OUTER" : joinType,
+          TABLE_1.text()
```

Comment:
If we pass in `TABLE_1` as last argument, should it not crash, as a stream-table join should not have a `WITHIN` clause?

Reply:
I just fixed the deprecation checker and changed this parameter.